x-pack/plugins/infra/server/lib/alerting/log_threshold/log_threshold_executor.ts

```diff
@@ -985,3 +1103,26 @@ export const FIRED_ACTIONS: ActionGroup<'logs.threshold.fired'> = {
     defaultMessage: 'Fired',
   }),
 };
+
+export const getIncludeSet = (
+  groupBy: string | string[] | undefined,
+  validPrefix: string[]
+): Set<string> => {
+  const includesList = [];
+  if (groupBy) {
+    if (Array.isArray(groupBy)) {
+      groupBy.forEach((group) => {
+        const groupPrefix = group.split('.')[0];
+        if (validPrefix.includes(groupPrefix)) {
+          includesList.push(groupPrefix);
+        }
+      });
+    } else {
+      const groupPrefix = groupBy.split('.')[0];
+      if (validPrefix.includes(groupPrefix)) {
+        includesList.push(groupPrefix);
+      }
+    }
+  }
+  return new Set<string>(includesList);
```

Comment:
nit: What do you think of expressing it like this so we don't need two branches?

```suggestion
  return new Set<string>(
    [groupBy ?? []]
      .flat()
      .map((currentGroupBy) => currentGroupBy.split('.')[0])
      .filter((groupByPrefix) => validPrefix.includes(groupByPrefix))
  );
```

Reply:
Good idea. Done 👍